terraform/environment.go

```diff
@@ -77,17 +95,67 @@ func (env *TerraformEnvironment) runCommand(args ...string) error {
 	return err
 }
 
+func getCmdOutputAndLog(cmd *exec.Cmd) ([]byte, error) {
+	var stdoutBuf, stderrBuf bytes.Buffer
+	stdoutIn, _ := cmd.StdoutPipe()
+	stderrIn, _ := cmd.StderrPipe()
+
+	var errStdout, errStderr error
+	debugLogWriterStdOut := newLogrusWriter(logrus.StandardLogger().WithField("Terraform", "StdOut"), logrus.DebugLevel)
+	defer debugLogWriterStdOut.Close()
+	debugLogWriterStdErr := newLogrusWriter(logrus.StandardLogger().WithField("Terraform", "StdErr"), logrus.DebugLevel)
+	defer debugLogWriterStdErr.Close()
+	stdout := io.MultiWriter(debugLogWriterStdOut, &stdoutBuf)
+	stderr := io.MultiWriter(debugLogWriterStdErr, &stderrBuf)
+	err := cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		_, errStdout = io.Copy(stdout, stdoutIn)
+		wg.Done()
+	}()
+
+	_, errStderr = io.Copy(stderr, stderrIn)
+	wg.Wait()
+
+	err = cmd.Wait()
+	if err != nil {
+		return nil, err
+	}
+	if errStdout != nil || errStderr != nil {
+		return nil, fmt.Errorf("failed to capture stdout or stderr\n")
+	}
+
+	if len(stderrBuf.Bytes()) > 0 {
+		err = fmt.Errorf(string(stderrBuf.Bytes()))
+	}
+
+	return stdoutBuf.Bytes(), err
+}
+
 func (env *TerraformEnvironment) runCommandResult(args ...string) ([]byte, error) {
-	if _, err := exec.LookPath(terraformCommand); err != nil {
+	if _, err := exec.LookPath(env.parameters.TerraformPath); err != nil {
 		return nil, errors.Wrap(err, "Terraform not installed. Please install terraform.")
 	}
 
-	cmd := exec.Command(terraformCommand, args...)
+	cmd := exec.Command(env.parameters.TerraformPath, args...)
 	cmd.Dir = env.WorkingDirectory
-	output, err := cmd.CombinedOutput()
+	logrus.Debugf("Running command: [%v] with args: %v", env.parameters.TerraformPath, args)
+	var err error
+	var output []byte
+	if env.parameters.Verbose {
+		output, err = getCmdOutputAndLog(cmd)
```

Comment:
nit: would it just make sense to always do this, but have the `verbose` flag trigger debug level?

Reply:
@lieut-data you  are absolutely right, no need for the 'else' branch. Fixed.